### PR TITLE
Possibility to turn distractions on and off

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Interface "Possibility to turn distractions on and off"

#### Purpose of change
Judging by the number of angry posts in reddit, forum, and other places, we need to add players a way to turn annoying distractions on and off.

#### Describe the solution
- Created a new distractions manager;
- The manager itself can be called through MAIN MENU ingame;
- Added a fourth option to distraction interrupt.

I deliberately didn't add switches for the following distraction types:
- `portal_storm_popup` as it's used only when talking to NPC and won't interrupt activities;
- `eoc` as it should happen silently and unavoidable;
- `noise` from your alarm clock as you turned the alarm by yourself and you _wanted_ to be interrupted;
- `motion_alarm` - the same as with `noise` above

#### Describe alternatives you've considered
- Integrate this stuff into Safemode manager;
- Make some options accept numbers instead of being simply on/off. For example, allow player enter value for hunger/thirst threshold.

#### Testing
Got myself `Antique Adornments` book. Spawned debug monster nearby. Tried to read the book with `Hostile is dangerously close` and `Hostile spotted` options turned on and off.

#### Additional context
изображение

изображение

изображение

### CODE REVIEW

**Comment by GuardianDll** [comment]
Won't it cause the angry posts for "i was killed cause the game did not spot the danger"? I can understand why it was implemented, but is it a good way? Can we make it also work as safe mode - add a button to a menu with "`Y`es, `N`o,"`I`gnore, `S`upress", and pressing `S` will prevent the similar messages for some time (maybe some fixed time or till the activity ends)?

**Comment by Night-Pryanik** [comment]
> Won't it cause the angry posts for "i was killed cause the game did not spot the danger"?

It would be completely the player's fault. You forgot to turn distraction back on? Blame yourself.

Making it behave like a safe mode with a "auto-turn-on after some period" mechanics is a good thing to have, but it would require way more work to implement than the current version. This enhancement could be added in subsequent PRs.

**Comment by anothersimulacrum** [comment]
This is a lot of new options. I think it would be better to have a menu like the safe mode manager for this sort of thing instead of adding so many new options.

**Comment by Night-Pryanik** [comment]
I agree, but making a new manager is far more difficult (for me) and time-consuming work , while Options menu already has all required infrastructure, so I went the easy way.

**Comment by Night-Pryanik** [comment]
Ok, I created a distraction manager. This is my first time that I created something complex like this, so please forgive my bad programming. I know it's ugly because everything is hardcoded and inefficient, but at least it works.

- Values for distraction types are now stored in `uistate` stuff;
- The manager itself can be called through MAIN MENU ingame
изображение
- Also I added a fourth option to distraction interrupt:
изображение
- The manager is pretty simple:
изображение

File: src/distraction_manager.cpp [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        // Draw border
+        draw_border( w_border, BORDER_COLOR, _( "Distractions manager" ) );
+        mvwputch( w_border, point( 0, iHeaderHeight - 1 ), c_light_gray, LINE_XXXO );
+        mvwputch( w_border, point( 79, iHeaderHeight - 1 ), c_light_gray, LINE_XOXX );
+        mvwputch( w_border, point( 61, FULL_SCREEN_HEIGHT - 1 ), c_light_gray, LINE_XXOX );
+        wnoutrefresh( w_border );
+
+        // Draw header
+        werase( w_header );
+        fold_and_print( w_header, point_zero, getmaxx( w_header ), c_white, string_format( _( "%s" ),
```
Comment (BevapDin) [inline src/distraction_manager.cpp:94]:
```suggestion
        fold_and_print( w_header, point_zero, getmaxx( w_header ), c_white,
```
```suggestion
        fold_and_print( w_header, point_zero, getmaxx( w_header ), c_white,
```
Translation the string "%s" is pretty much pointless, as is formatting with it. And if you absolutely want it, `fold_and_print` can do the formatting for you.

File: src/distraction_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            } else {
+                ++currentLine;
+            }
+        } else if( currentAction == "CONFIRM" ) {
+            // This will change status color and status text
+            distractions_status[currentLine] = !distractions_status[currentLine];
+
+            // This will actually toggle enabled/disabled status of distractions
+            switch( currentLine ) {
+                case 0:
```
Comment (ZhilkinSerg) [inline src/distraction_manager.cpp:170]: It could be worth converting these numbers into enums.

Comment (Night-Pryanik) [reply src/distraction_manager.cpp:170]: I thought about it. The thing is, there is already one enum with all distraction types, I just didn't want to create another one, almost exactly the same as original, just without few types.

File: src/distraction_manager.cpp [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        calcStartPos( startPosition, currentLine, iContentHeight, number_of_distractions );
+
+        for( int i = startPosition; i < number_of_distractions; ++i ) {
+            const nc_color line_color = i == currentLine ? hilite( c_white ) : c_white;
+            const nc_color status_color = distractions_status[i] ? c_light_green : c_red;
+            const std::string status_string = distractions_status[i] ? _( "Enabled" ) : _( "Disabled" );
+
+            // Print distraction types
+            mvwprintz( w, point( 1, i - startPosition ), line_color, "%s",
+                       configurable_distractions[i].first.c_str() );
```
Comment (BevapDin) [inline src/distraction_manager.cpp:132]:
```suggestion
                       _( configurable_distractions[i].first.c_str() ) );
```
I think this is required for translation to work. The string is marked for extraction, but it is not actually translated anywhere.

**Pull request merged**